mitosheet/mitosheet/updates/replay_analysis.py

```diff
@@ -6,27 +6,79 @@
 """
 Replays an existing analysis
 """
-
+from copy import copy, deepcopy
+from typing import Any, Dict, List
 from mitosheet.errors import make_no_analysis_error
 
 from mitosheet.telemetry.telemetry_utils import log
 from mitosheet.saved_analyses import read_and_upgrade_analysis
 from mitosheet.types import StepsManagerType
+from mitosheet.step_performers.import_steps import is_import_step_type
+from mitosheet.api.get_imported_files_and_dataframes_from_current_steps import get_import_data_with_single_import_list
 
 REPLAY_ANALYSIS_UPDATE_EVENT = 'replay_analysis_update'
 REPLAY_ANALYSIS_UPDATE_PARAMS = [
     'analysis_name',
+    'step_import_data_list_to_overwrite'
 ]
 
+def overwrite_import_data(analysis: Dict[str, Any], step_import_data_list_to_overwrite: List[Dict[str, Any]]) -> Dict[str, Any]:
+    """
+    If there is import data to overwrite, we go through and overwrite it. We notably have to do this
+    by keeping track of the index of the import, as we do not have step ids to link these things. 
+
+    As this import data is split so that only one dataframe is created during any specific import,
+    this means we make a bit more steps than we might want. But it's ok for now!
+    """
+    
+    # Flatten all the imports into a single list, so that we can index into them. We notably
+    # don't need the step ids for these
+    all_imports = [_import for import_data in step_import_data_list_to_overwrite for _import in import_data['imports']]
```

Comment:
TODO: let's take this code, refactor into a utility, and use it anywhere we need to (e.g. anywhere we manually track and index).